packages/sit-onyx/src/components/OnyxButton/OnyxButton.vue

```diff
@@ -0,0 +1,265 @@
+      resultColor = "var(--onyx-color-text-icons-primary-intense, #0079CE)";
+    } else if (props.color === "secondary") {
+      resultColor = "var(--onyx-color-text-icons-neutral-intense, #52626D)";
+    } else if (props.color === "danger") {
+      resultColor = "var(--onyx-color-text-icons-danger-intense, #D1332F)";
+    }
+  }
+
+  return resultColor;
+});
+
+const disabledColor = computed(() => {
+  let resultColor = "transparent";
+
+  if (props.variant === "default" && props.color === "primary") {
+    resultColor = "var(--onyx-color-text-icons-neutral-inverted, #FFF)";
+  } else {
+    if (props.color === "primary") {
+      resultColor = "var(--onyx-color-text-icons-primary-soft, #BFDDF3)";
+    } else if (props.color === "secondary") {
+      resultColor = "var(--onyx-color-text-icons-neutral-soft, #95A1AA)";
+    } else if (props.color === "danger") {
+      resultColor = "var(--onyx-color-text-icons-danger-medium, #E6A7A6)";
+    }
+  }
+
+  return resultColor;
+});
+
+const borderColor = computed(() => {
+  let resultColor = "transparent";
+
+  if (props.variant === "default" || props.variant === "outline") {
+    if (props.color === "primary") {
+      resultColor = "var(--onyx-color-base-primary-500, #0079ce)";
+    } else if (props.color === "secondary") {
+      resultColor = "var(--onyx-color-base-neutral-400, #B7BFC5)";
+    } else if (props.color === "danger") {
+      resultColor = "var(--onyx-color-base-danger-500, #D1332F)";
+    }
+  } else {
+    resultColor = "transparent";
+  }
+
+  return resultColor;
+});
+
+const disabledBorderColor = computed(() => {
+  let resultColor = "transparent";
+
+  if (props.variant === "default" || props.variant === "outline") {
+    if (props.color === "primary") {
+      resultColor = "var(--onyx-color-base-primary-200, #BFDDF3)";
+    } else if (props.color === "secondary") {
+      resultColor = "var(--onyx-color-base-neutral-200, #EAEDF0)";
+    } else if (props.color === "danger") {
+      resultColor = "var(--onyx-color-base-danger-200, #F1D3D2)";
+    }
+  } else {
+    resultColor = "transparent";
+  }
+
+  return resultColor;
+});
+
+const focusOutlineColor = computed(() => {
+  let resultColor = "transparent";
+
+  if (props.color === "primary") {
+    resultColor = "var(--onyx-color-base-primary-200, #BFDDF3)";
+  } else if (props.color === "secondary") {
+    resultColor = "var(--onyx-color-base-neutral-300, #D8DDE1)";
+  } else {
+    resultColor = "var(--onyx-color-base-danger-300, #E6A7A6)";
+  }
+
+  return resultColor;
+});
+</script>
+
+<template>
+  <div
+    tabindex="0"
+    class="onyx-button"
+    :class="{
+      'onyx-button--disabled': props.isDisabled,
+    }"
+  >
+    <button
+      v-bind="props"
+      ref="buttonElement"
+      :disabled="props.isDisabled"
+      class="onyx-button__element"
+      @click="handleClick"
+    >
+      <span class="onyx-button__element__label">{{ props.label }}</span>
+    </button>
+  </div>
```

Comment:
Can I add an icon inside the button element?

I wrapped it in a div because of this structure in Figma

![image](https://github.com/SchwarzIT/onyx/assets/60856270/f9a2d4a7-31eb-4a92-8f47-a4de68a32879)

Reply:
We can make use of pseudo-classes: E.g. In this https://developer.mozilla.org/en-US/docs/Web/CSS/::before

Reply:
The icon is probably implemented with our `OnyxIcon` component so we can not really use ::before.

But we can just place the icon inside the `<button>` like so:
```html
<button>
  <OnyxIcon />
  <span class="onyx-button__element__label">{{ props.label }}</span>
</button>
```

Like this we can also get rid of the div :)